### ISSUE TITLE
Zero export limit

### ISSUE THREAD

**Issue by djfanatix** [biemond/solaredge.modbus#12]
Hi,

Could you add a option to read and write the export limit?
It is address F001 

[[](url)](https://tweakers.net/i/_Vbw6h07uP3EtmVy-RP_ZXr0jnM=/full-fit-in/4000x4000/filters:no_upscale():fill(white):strip_exif()/f/image/IHKtvheNjManYE9TaabWoDh8.png?f=user_large)

Thanks

**Comment by biemond** [comment]
yeah, will do, should be easy.

export limit is what we control or not?


By the way for your solcast,  is it possible to add a flow condition so I can charge the batt with dal tarief when needed (like when expected kwh for today is less ) , I run this after midnight so prediction should be in for that day..

**Comment by biemond** [comment]
can you test this and see if this is what you want.

https://homey.app/nl-nl/app/solaredge.modbus/SolarEdge/test/

and https://github.com/biemond/solaredge.modbus/commit/80bd9ae7780b22c7404ca49af1f91bd3ff6b8dae

<img width="819" alt="Screen Shot 2022-06-14 at 23 36 06" src="https://user-images.githubusercontent.com/1085004/173693346-2ba74076-2da9-4597-8c75-f414750fce19.png">

**Comment by djfanatix** [comment]
Yes I see the same.

I found some more documentation:
[](https://user-images.githubusercontent.com/2674434/134251823-c404b894-037e-469b-bc9b-8787ebc1770a.png)


Can you make a flow card or a setting to change the power limit?

I think people with dynamic energy contracts can then enable zero export in case of negative prices

**Comment by biemond** [comment]
really that is a thing in europe.  you have to pay when you do export to the grid. So you can only do this when you have batt or when it is stimulated and get paid for it.

**Comment by biemond** [comment]
so an enable/disable flow with.   advanced power on 1 and reactive power config to 4 or for normal/enabled both back to 0 
is this also a menu setting/action? which one?

if you do this on the invertor = active power limit 0 % then?

**Comment by djfanatix** [comment]
> really that is a thing in europe. you have to pay when you do export to the grid. So you can only do this when you have batt or when it is stimulated and get paid for it.

No most of the time you get paid to export, but on sunny and windy days, the rates are negative, so you have to pay.
(or you get paid to import)

**Comment by djfanatix** [comment]
I found this in another repo:
I think I was wrong, the active power limit, limits the inverter, that's not what we need, we need to limit export (with a meter of course)

I don't think (for my needs) that E001 has to be included.

#0xE000 - 1 - Export control mode: 16bit_uint
     0: "Disabled",
     1: "Direct Export Limitation",
     2: "Indirect Export Limitation",
     4: "Production Limitation"
#0xE001 - 1 - Export control limit mode:16bit_uint
     0: "Total",
     1: "Per phase"
#0xE002 - 2 - Export control site limit: 32bit_float
["Export control site limit", "export_control_site_limit", 0xE002, "f", {"min": 0, "max": 10000, "unit": "W"}],

**Comment by biemond** [comment]
yes that looks better , I know sometimes on US or AUS you cannot deliver back to the grid.

**Comment by djfanatix** [comment]
> 
> By the way for your solcast, is it possible to add a flow condition so I can charge the batt with dal tarief when needed (like when expected kwh for today is less ) , I run this after midnight so prediction should be in for that day..

I checked this, but I am not planning to rewrite the app for sdk 3, as it too much work with the whole UTC and crontasks thing.
You can just use logic and the solcast tag and use it as a condition. (greater or smaller then..)

**Comment by biemond** [comment]
thanks

**Comment by biemond** [comment]
I need this from oct/nov

**Comment by biemond** [comment]
<img width="1179" alt="Screen Shot 2022-06-15 at 20 59 25" src="https://user-images.githubusercontent.com/1085004/173904655-1420a629-1c32-4855-834f-26965d9f80d3.png">

<img width="1177" alt="Screen Shot 2022-06-15 at 21 00 13" src="https://user-images.githubusercontent.com/1085004/173904569-a0e3fd1b-db1f-41ce-981a-2ca1216130de.png">

**Comment by biemond** [comment]
so only when you have a meter.

**Comment by biemond** [comment]
```
        "advancedpwrcontrolen": [0xf142, 2, 'UINT32', "Advanced Power Control En"],
        "reactivepwrconfig": [0xf102, 2, 'UINT32', "Reactive Power Config"],        
        "export_control_mode": [0xe000, 1, 'UINT16', "Export control Mode"],
        "export_control_limit_mode": [0xe001, 1, 'UINT16', "Export control limit Mode"],
        "export_control_site": [0xe002, 2, 'FLOAT32', "Export control site limit"],
        "powerreduce": [0xf140, 2, 'FLOAT32', "Power Reduce"],
        "maxcurrent": [0xf18e, 2, 'FLOAT32', "Max Current"]


rrcr_state 0 xxx RRCR State
active_power_limit 100 xxx Active Power Limit
cosphi 1 xxx CosPhi
advancedpwrcontrolen 1 xxx Advanced Power Control En
reactivepwrconfig 4 xxx Reactive Power Config
export_control_mode 0 xxx Export control Mode
export_control_limit_mode 0 xxx Export control limit Mode
export_control_site -3.4028234663852886e+38 xxx Export control site limit
powerreduce 100 xxx Power Reduce
maxcurrent 7071.06787109375 xxx Max Current
```

**Comment by biemond** [comment]
tested it on 5000
```
rrcr_state 0 xxx RRCR State
active_power_limit 100 xxx Active Power Limit
cosphi 1 xxx CosPhi
advancedpwrcontrolen 1 xxx Advanced Power Control En
reactivepwrconfig 4 xxx Reactive Power Config
export_control_mode 1 xxx Export control Mode
export_control_limit_mode 0 xxx Export control limit Mode
export_control_site 5000 xxx Export control site limit
powerreduce 100 xxx Power Reduce
maxcurrent 7071.06787109375 xxx Max Current
```
<img width="1003" alt="Screen Shot 2022-06-20 at 19 14 52" src="https://user-images.githubusercontent.com/1085004/174652334-eaeb8dfb-77cb-40df-b4e9-4a50cf1bfe8a.png">
<img width="1028" alt="Screen Shot 2022-06-20 at 19 15 13" src="https://user-images.githubusercontent.com/1085004/174652353-e9e49c4d-2c49-4443-8c8f-3de9140208ff.png">

<img width="1079" alt="Screen Shot 2022-06-20 at 19 16 46" src="https://user-images.githubusercontent.com/1085004/174652373-72d14d2f-bb63-405d-b326-65889f124ed7.png">
<img width="1036" alt="Screen Shot 2022-06-20 at 19 16 57" src="https://user-images.githubusercontent.com/1085004/174652385-ad19eb5b-8b34-4dda-b5f9-0e80d320c1da.png">
<img width="1110" alt="Screen Shot 2022-06-20 at 19 17 13" src="https://user-images.githubusercontent.com/1085004/174652398-5fcccb95-5d95-4e43-a28a-94c95425298b.png">

**Comment by biemond** [comment]
I think I only need to set these 3
```
export_control_mode 1 xxx Export control Mode
export_control_limit_mode 0 xxx Export control limit Mode  , already on site / not phase.
export_control_site 5000 xxx Export control site limit
```

**Comment by biemond** [comment]
can you try this https://homey.app/a/solaredge.modbus/test/
for now only a menu item ( no flow ) and export limit fixed on 5000W

<img width="428" alt="Screen Shot 2022-06-21 at 00 46 44" src="https://user-images.githubusercontent.com/1085004/174686077-68f35303-c3f3-44cd-bfa5-6a1e21ebce9f.png">
<img width="444" alt="Screen Shot 2022-06-21 at 00 46 59" src="https://user-images.githubusercontent.com/1085004/174686090-345bc747-ba56-49d8-8caf-2eec563ead29.png">

**Comment by djfanatix** [comment]
Looks good, my site limit is 4500w, so I can’t test it at the moment.

**Comment by biemond** [comment]
ok , 4500 is your energy provider restriction or do you get some error on solaredge. 

shall I change it to 4500 to find out if it works or give me some value.

**Comment by djfanatix** [comment]
no, I only have a small inverter, 4kw.
Can you try 3000 or so?

**Comment by biemond** [comment]
yes , let me do this now.

**Comment by biemond** [comment]
done https://homey.app/a/solaredge.modbus/test/

**Comment by djfanatix** [comment]
![image](https://user-images.githubusercontent.com/43752712/174749741-75c8dcc0-c6a2-4648-a211-6cb714d4a8e8.jpeg)

Works! Thank you.

I double checked in HA and I see the same data

**Comment by biemond** [comment]
beter, mooi.  I will work on the flows and this export limit setting

**Comment by biemond** [comment]
and the flows https://homey.app/a/solaredge.modbus/test/ plus you can change export limit watts

**Comment by MTRKhn** [comment]
Hi,

could you please consider adding a 0W export limit setting ? I can’t find a way to enable zero export :(

thanks !

**Comment by biemond** [comment]
oh , let me check

**Comment by biemond** [comment]
@MTRKhn can you try this one https://homey.app/a/solaredge.modbus/test/

**Comment by MTRKhn** [comment]
UI seems ok. Now the challenge will be to generate enough to check if it’s exporting or not 😅

![060EF19E-FE43-4630-8090-8969F12A3521](https://user-images.githubusercontent.com/72077818/205486765-cf251b21-3f0b-449d-9f77-c75e0c89b4fa.jpeg)

**Comment by biemond** [comment]
yes winter time.  it is hard to get something and go over your own consumption
